### ISSUE TITLE
type 2: add vocabulary and embedding layer 

### ISSUE THREAD

**Issue by anupamme** [anupamme/keras-triplet-loss-mnist#2]
If we have 2 random variables A and B which are uniform between 0 and 1. Then we sample values of A and B as many times, this way we have X of the input: X = (A, B). To calculate Y, we need a constraint X must satisfy e.g.

A^2 + B^2 < 0.5

So for values of A, B the constraint is satisfied, Y is 1 else it is 0.

The possible values of A and B are infinite but we cannot have an infinite vocabulary for the embedding. To solve this issue: if our vocabular size is 10K, then we limit A and B to 4 digits after 0 because total possible values of A and B are 10K in that case.

Mapping to vocabulary: once we have values of A e.g. 0.1234 we need to map it to 4 digit vocabulary we simply multiply by 10K so in this case 0.1234 gets mapped to 1234.

So we have a function **convert_to_vocab** which does the mapping.

To add embedding layer in the network we add two layers to FF 1 layer so now FF looks like:

 tf.keras.layers.Embedding(10000, 128, input_length=2), # assuming 2 dimentional input A and B.
 tf.keras.layers.Flatten(),
tf.keras.layers.Dense(256, activation=None)

Some issues on this:

1. In these two close numbers e.g. 0.1234 and 0.1235 get mapped to IDs which are close 1234, 1235 but we are not introducing any bias to ensure that these should be close in embedding space.

2. If an ID is not present in training its embedding will be set to a random value so it will give incorrect value in the test case.

3. Can you create embeddings for pair e.g. (0.1234, 0.2345) gets an ID rather than 0.1234 and 0.2345 getting their individual IDs.
